internal/pkg/daemon/bpfrecorder/bpf/recorder.bpf.c

```diff
@@ -165,13 +166,26 @@ static __always_inline int register_file_event(struct file * file, u64 flags)
         return 0;
     }
 
-    int ok = bpf_d_path(&file->f_path, event->data, sizeof(event->data));
-    if (ok < 0) {
-        bpf_printk("register_file_event bpf_d_path failed: %i\n", ok);
+    int pathlen = bpf_d_path(&file->f_path, event->data, sizeof(event->data));
+    if (pathlen < 0) {
+        bpf_printk("register_file_event bpf_d_path failed: %i\n", pathlen);
         bpf_ringbuf_discard(event, 0);
         return 0;
     }
 
+    if (file->f_inode->i_mode & S_IFDIR) {
+        // overly pedantic check to make ebpf verifier happy
+        if (pathlen - 2 < sizeof(event->data) && pathlen - 1 < sizeof(event->data) && pathlen < sizeof(event->data)){
+            if(event->data[pathlen - 2] != '/') {
```

Comment:
I think we need to check if this is also grater than 0. It might crash when the index is -1. e.g. when pathlen is 1.

Reply:
I think the other two checks `pathlen - 1` < and `pathlen <` are redundant because they are already implicit when the pathlen - 2 is smaller than sizeof(event->data).

Reply:
They are redundant, but if we leave them out the eBPF verifier doesn't let me load the program. I've rephrased the comment to make this more clear. And added the minimum size check, good call. 🙂